frontend/journal.js

```diff
@@ -8,12 +8,79 @@ let listAlways = new Array();
 let date = new Date();
 
 date = today();
+
+//returns the day of the week between 1-7
+function day(strDate) {
+    let temp = new Date(Date.parse(strDate));
+    return temp.getDay();
+}
+
+function getBeginningOfTheWeek(strDate) {
+    const dateTmp = new Date(strDate);
+    let day = dateTmp.getDay();
+    let diff = dateTmp.getDate() - day + (day == 0 ? -6 : 1); // adjust when day is sunday
+    return dateToString(new Date(dateTmp.setDate(diff)));
+}
+
+let listWeek = new Array();
```

Comment:
better have constant array and push and slice element to it instead of using array[i] = n to update array.